### ISSUE TITLE
"not-callable" reported with contextmanager yields a callable

### ISSUE THREAD

**Issue by kstrauser** [PyCQA/pylint#1746]
### Steps to reproduce

Create a `contextmanager`-decorator function that yields a nested `contextmanager`. For example:

```python
from contextlib import contextmanager

@contextmanager
def outer():
    print('Entering outer')
    @contextmanager
    def inner():
        print('Entering inner')
        yield
        print('Leaving inner')

    yield inner

    print('Leaving outer')


with outer() as foo:
    with foo():  # E: 18, 9: foo is not callable (not-callable)
        print('In the zone')
```

### Current behavior

pylint 1.7.4 incorrectly reports the nested context manager as `not-callable`.

### Expected behavior

pylint would recognize that after `with outer() as foo:`, `foo` is in fact a valid callable.

### pylint --version output

```
No config file found, using default configuration
pylint 1.7.4,
astroid 1.5.3
Python 3.6.3 (default, Oct  4 2017, 06:09:38)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

**Comment by PCManticore** [comment]
Thank you for reporting this bug!

**Comment by brycepg** [comment]
Looks like, for some reason astroid receives the value from the 'first' yield instead of the yield in the proper scope

    code = """
    from contextlib import contextmanager

    @contextmanager
    def outer():
        @contextmanager
        def inner():
            yield 2
        yield inner

    with outer() as foo:
        foo #@
    """
    astroid.extract_node(code).inferred()[0].value

returns `2` not the function

**Comment by kstrauser** [comment]
Woo-hoo! Thanks so much, @brycepg!
